src/store/indexeddb-local-backend.ts

```diff
@@ -401,11 +403,22 @@ export class LocalIndexedDBStoreBackend implements IIndexedDBBackend {
     public async syncToDatabase(userTuples: UserTuple[]): Promise<void> {
         const syncData = this.syncAccumulator.getJSON(true);
 
+        if (this.isPersisting) {
+            logger.warn("Skipping syncToDatabase() as persist already in flight");
+            this.pendingUserPresenceData.push(...userTuples);
+            return;
+        } else {
+            userTuples.unshift(...this.pendingUserPresenceData);
+            this.isPersisting = true;
+        }
```

Comment:
given the return we could outdent this `else` case

Reply:
*shrug* - the intent is to explicitly show the alternation. not sure what benefit there is to outdenting it, other than make the flow of control harder to grok.